src/benchmarking/randomized_benchmarking.jl

```diff
@@ -0,0 +1,342 @@
+    return RandomizedBenchmarkingFitProperties(order, initial_parameters)
+end
+
+function get_sequence_fidelities(simulate_shots, transpile!,
+    properties::RandomizedBenchmarkingProperties)
+
+    circuit_list = QuantumCircuit[]
+    for i_length in 1:length(properties.sequence_length_list)
+        circuit_list_for_length =
+            get_random_clifford_circuits(i_length, transpile!, properties)
+        append!(circuit_list, circuit_list_for_length)
+    end
+    
+    shots_list = simulate_shots(circuit_list)
+    sequence_fidelity_list = []
+    circuit_id = 1
+    for i_length in 1:length(properties.sequence_length_list)
+        survival_probability_sum = 0
+        num_circuits_for_length = properties.num_circuits_per_length[i_length]
+        for j_circuit_for_length = 1:num_circuits_for_length
+            survival_probability_sum += get_survival_probability(shots_list[circuit_id],
+                properties.target_qubits)
+            circuit_id += 1
+        end
+        sequence_fidelity = survival_probability_sum/num_circuits_for_length
+        push!(sequence_fidelity_list, sequence_fidelity)
+    end
+    return sequence_fidelity_list
+end
+
+function get_survival_probability(shots, target_qubits)
+    num_successes = 0
+    num_target_qubits = length(target_qubits)
+    for measurement in shots
+        trimmed_measurement = measurement[target_qubits]
+        if trimmed_measurement == "0"^num_target_qubits
+            num_successes += 1
+        end
+    end
+    survival_probability = num_successes/length(shots)
+    return survival_probability
+end
+
+function get_random_clifford_circuits(sequence_length_id, transpile!,
+    properties::RandomizedBenchmarkingProperties)
+
+    circuit_list = []
+    num_target_qubits = length(properties.target_qubits)
+    qubit_map = Dict(zip(1:num_target_qubits, properties.target_qubits))
+
+    for i_circuit in 1:properties.num_circuits_per_length[sequence_length_id]
+        old_clifford = get_random_clifford(num_target_qubits)
+        circuit = get_transpiled_circuit(old_clifford, properties, qubit_map, transpile!)
+        clifford_product = old_clifford
+
+        for j_length = 2:properties.sequence_length_list[sequence_length_id]
+            new_clifford = get_random_clifford(num_target_qubits)
+            new_circuit = get_transpiled_circuit(new_clifford, properties, qubit_map,
+                transpile!)
+            append!(circuit, new_circuit)
+            clifford_product = clifford_product*new_clifford
+            old_clifford = new_clifford
+        end
+        inverse_clifford = inv(clifford_product)
+        inverse_circuit = get_transpiled_circuit(inverse_clifford, properties,
+            qubit_map, transpile!)
+        append!(circuit, inverse_circuit)
+        push!(circuit_list, circuit)
+    end  
+    return circuit_list
+end
+
+function get_transpiled_circuit(clifford, properties, qubit_map, transpile!)
```

Comment:
same as above. Signatures like this are very scary for another developer and hard to understand.